src/analysis/normed_space/star/complex.lean

```diff
@@ -0,0 +1,56 @@
+/-
+Copyright (c) 2022 Frédéric Dupuis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Dupuis
+-/
+
+import analysis.normed_space.star.basic
+import algebra.star.module
+import analysis.complex.basic
+
+/-!
+# Complex normed star modules and algebras
+
+Facts about star modules and star algebras over the complex numbers.
+
+## Main definitions
+
+* `star_module.mul_neg_I_lin`: multiplication by -I as a real-linear equivalence between the
+  skew-adjoint and self-adjoint elements of a star module.
+* `star_module.im`: the imaginary part of an element of a star module, defined via
+  `skew_adjoint_part`.
+
+-/
+
+variables {E : Type*}
+
+namespace star_module
+open_locale complex_conjugate
+open complex
+
+variables [add_comm_group E] [star_add_monoid E] [module ℂ E] [star_module ℂ E]
+
+/-- Multiplication by -I as a real-linear equivalence between the skew-adjoint and self-adjoint
+elements of a star module. -/
+@[simps] def mul_neg_I_lin : skew_adjoint E ≃ₗ[ℝ] self_adjoint E :=
+{ to_fun := λ x, ⟨-I • x, by simp [self_adjoint.mem_iff]⟩,
+  inv_fun := λ x, ⟨I • x, by simp [skew_adjoint.mem_iff]⟩,
+  map_add' := λ x y, by { ext, simp only [add_subgroup.coe_add, smul_add, add_subgroup.coe_mk] },
+  map_smul' := λ r x, by { ext, simp only [neg_smul, neg_inj, skew_adjoint.coe_smul,
+    add_subgroup.coe_mk, ring_hom.id_apply, self_adjoint.coe_smul, smul_neg, smul_comm I], },
+  left_inv := λ x, by simp only [neg_smul, add_subgroup.coe_mk, smul_neg, ←mul_smul, I_mul_I,
+                                 neg_neg, one_smul, set_like.eta],
+  right_inv := λ x, by simp only [←mul_smul, I_mul_I, add_subgroup.coe_mk, neg_mul, neg_neg,
+                                  one_smul, set_like.eta] }
```

Comment:
You could also define this as
```suggestion
(linear_equiv.of_submodules
  (distrib_mul_action.to_linear_equiv ℝ E (⟨-I, I, by simp, by simp⟩ : ℂˣ))
  { smul_mem' := λ c x, skew_adjoint.smul_mem c, ..skew_adjoint E }
  { smul_mem' := λ c x, self_adjoint.smul_mem c, ..self_adjoint E }
  begin
    ext,
    simp [units.smul_def, self_adjoint.mem_iff, skew_adjoint.mem_iff, ←inv_smul_eq_iff₀ (I_ne_zero),
      ←mul_smul],
  end : _ ≃ₗ[ℝ] _)
```
but it's unfortunately a lot slower. It would be a bit less painful if we had a `skew_adjoint.submodule` etc viewing these add_subgroups as submodules, but I think what you have right now is fine.

Reply:
I agree having a way to convert to submodules would be a good idea, but let's leave that for a future PR!